src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/ReInviteUserProfileIntTest.java

```diff
@@ -0,0 +1,173 @@
+
+        UserProfileCreationData data = buildCreateUserProfileData(true);
+        ErrorResponse errorResponse = (ErrorResponse) createUser(data, NOT_FOUND, ErrorResponse.class);
+        assertThat(errorResponse.getErrorMessage()).isEqualTo("4 : Resource not found");
+        assertThat(errorResponse.getErrorDescription()).contains("could not find user profile");
+    }
+
+    // AC4: resend invite to a given user who is not in the 'Pending' state
+    @Test
+    public void should_return_400_when_user_reinvited_is_not_pending() throws Exception {
+
+        userProfile.setStatus(IdamStatus.ACTIVE);
+        userProfileRepository.save(userProfile);
+        updateLastUpdatedTimestamp(userProfile.getIdamId());
+
+        UserProfileCreationData data = buildCreateUserProfileData(true);
+        data.setEmail(pendingUserRequest.getEmail());
+        ErrorResponse errorResponse = (ErrorResponse) createUser(data, BAD_REQUEST, ErrorResponse.class);
+        assertThat(errorResponse.getErrorMessage()).isEqualTo("3 : There is a problem with your request. Please check and try again");
+        assertThat(errorResponse.getErrorDescription()).isEqualTo("User is not in PENDING state");
+
+    }
+
+    // AC8: resend invite to a given user who was last invited less than 1 hour before
+    @Test
+    public void should_return_429_when_user_reinvited_within_one_hour() throws Exception {
+
+        UserProfileCreationData data = buildCreateUserProfileData(true);
+        data.setEmail(pendingUserRequest.getEmail());
+        ErrorResponse errorResponse = (ErrorResponse) createUser(data, TOO_MANY_REQUESTS, ErrorResponse.class);
+        assertThat(errorResponse.getErrorMessage()).isEqualTo(String.format("10 : The request was last made less than %s minutes ago. Please try after some time", resendInterval));
+        assertThat(errorResponse.getErrorDescription()).contains(String.format("The request was last made less than %s minutes ago. Please try after some time", resendInterval));
+
+    }
+
+    // AC9: invited more than an hour ago but has recently activated their account
+    @Test
+    public void should_return_409_when_reinvited_user_gets_active_in_sidam_but_pending_in_up() throws Exception {
+
+        updateLastUpdatedTimestamp(userProfile.getIdamId());
+        setSidamRegistrationMockWithStatus(HttpStatus.CONFLICT.value());
+
+        UserProfileCreationData data = buildCreateUserProfileData(true);
+        data.setEmail(pendingUserRequest.getEmail());
+        ErrorResponse errorResponse = (ErrorResponse) createUser(data, HttpStatus.CONFLICT, ErrorResponse.class);
+        assertThat(errorResponse.getErrorMessage()).isEqualTo(String.format("7 : Resend invite failed as user is already active. Wait for %s minutes for the system to refresh.", syncInterval));
+        assertThat(errorResponse.getErrorDescription()).contains(String.format("Resend invite failed as user is already active. Wait for %s minutes for the system to refresh.", syncInterval));
```

Comment:
This message need to change.. we should not have sync interval time  in the message as this opens up a possibility of UP sending a wrong message to exui if the sync interval timings are not synchronized btw profile sync and user profile

Reply:
is the same message used by Ex Ui to show to the end user?